bern-university-of-applied-sciences-institute-for-digital-economy-in-the-construction-and-wood-industry-idbh.csl

```diff
@@ -0,0 +1,341 @@
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="report thesis" match="any">
+        <text variable="title" font-style="italic"/>
+        <group prefix=" [" suffix="]" delimiter=" ">
+          <text variable="genre"/>
+          <text variable="number" prefix="No. "/>
+        </group>
+      </if>
+      <else-if type="bill book graphic legal_case legislation motion_picture report song speech" match="any">
+        <text variable="title" font-style="italic"/>
+        <text macro="edition" prefix=", "/>
+      </else-if>
+      <else-if type="webpage">
+        <text variable="title" font-style="italic"/>
+        <text value="online publication" prefix=" [" suffix="]"/>
+      </else-if>
+      <else-if type="post post-weblog">
+        <text variable="title" font-style="italic"/>
+        <text value="online forum" prefix=" [" suffix="]"/>
+      </else-if>
+      <else-if type="paper-conference">
+        <text variable="title" font-style="italic"/>
+        <text value="unpublished source" prefix=" [" suffix="]"/>
+      </else-if>
+      <else>
+        <text variable="title" font-style="italic"/>
+        <text variable="publisher" prefix=", "/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=", ">
+      <text variable="publisher"/>
+      <text variable="publisher-place"/>
+    </group>
+  </macro>
+  <macro name="event">
+    <choose>
+      <if variable="event">
+        <text term="presented at" text-case="capitalize-first" suffix=" "/>
+        <text variable="event"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="issued">
+    <choose>
+      <if variable="issued">
+        <date variable="issued">
+          <date-part name="year"/>
+        </date>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="edition">
+    <group delimiter=" ">
+      <choose>
+        <if is-numeric="edition">
+          <number variable="edition" form="ordinal"/>
+        </if>
+        <else>
+          <text variable="edition" suffix="."/>
+        </else>
+      </choose>
+      <text value="ed"/>
+    </group>
+  </macro>
+  <macro name="locators">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper" match="any">
+        <group prefix=" " delimiter=", ">
+          <group>
+            <text variable="volume"/>
+            <text variable="issue" prefix=" - edition "/>
+          </group>
+          <text variable="page" prefix="pages "/>
+        </group>
+      </if>
+      <else-if type="bill book graphic legal_case legislation motion_picture report song thesis" match="any">
+        <group delimiter=", " prefix=". ">
+          <text macro="event"/>
+          <text macro="publisher"/>
+        </group>
+      </else-if>
+      <else-if type="chapter paper-conference" match="any">
+        <group delimiter=", " prefix=". ">
```

Comment:
Throughout the style, you have a lot of punctuation in prefixes and suffixes. This creates high risk for extra punctuation being added when variables are missing. Please go through the style and replace these with `<group delimiter=". ">` and similar so that punctuation only appears when necessary.

Reply:
See here for a discussion and example: https://github.com/citation-style-language/styles/blob/master/STYLE_DEVELOPMENT.md#changing-punctuation